### PR TITLE
Handle Validate errors with unicode

### LINKED ISSUE #72: Invalid expression causes an exception
An invalid expression, such as `$(new_post_installNo)='2'` (incorrect parentheses) causes an exception such as -

```
Traceback (most recent call last):
  File "/Users/elvin/py/pyxform/xls2xform.py", line 94, in <module>
    main_cli()
  File "/Users/elvin/py/pyxform/xls2xform.py", line 81, in main_cli
    response['message'] = str(e)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u034e' in position 209: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by lognaturel**
Closes #72 

The test did not fail originally because the issue was with displaying the error. It seems like it's still good to have a test for it but I'm not sure if it's actually useful.

I manually verified that https://docs.google.com/spreadsheets/d/1gBJVFqbW0vK7B3cpVmZMPEdJ34ulf7PWiQAynxCo7Dk/edit#gid=0 originally failed and now works with these changes applied.

### CODE REVIEW

**Comment by lognaturel** [comment]
Hmm. Would be good for someone with more knowledge of Python 2 vs. 3 quirks to take a look here. Looks like `ErrorCleaner.odk_validate(stderr)` is a `str` in Python 3 so the original code wasn't a problem. But in Python 2 `ErrorCleaner.odk_validate(stderr)` is a `unicode`. My fix is for Python 2.

File: pyxform/validators/odk_validate/__init__.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def check_xform(path_to_xform):
     else:
         if returncode > 0:  # Error invalid
             raise ODKValidateError(
-                'ODK Validate Errors:\n' + ErrorCleaner.odk_validate(stderr))
+                'ODK Validate Errors:\n' + ErrorCleaner.odk_validate(stderr).encode('utf-8'))
```
Comment (ukanga) [inline pyxform/validators/odk_validate/__init__.py]: You could try
```python
b'ODK Validate Errors:\n' + ErrorCleaner.odk_validate(stderr).encode('utf-8'))
```
Works with both python 3 and python 2.

Comment (lognaturel) [reply pyxform/validators/odk_validate/__init__.py]: Aha, thank you, that makes a lot of sense! Now all tests are passing on Travis but it's not clear to me why they're failing on AppVeyor. 😣All tests for Python 2 and 3 now say
```
AssertionError: odk_validate_error__contains: Couldn't find u'Invalid calculate for the bind attached to "${bad}" : Couldn\'t understand the expression starting at this point: \u034e$(myField)=\'1\' in expression $(myField)=\'1\'' in content
```

Is it another type issue? But they why does it work on Travis?

**Comment by ukanga** [comment]
@lognaturel The reason the all windows tests are failing may have to do with ODK validate not returning a valid unicode character. I get a `?` character in the string and as such it is not an exact match to the test string. See the error below which I got from a windows 10 VM.
```java
Parsing form...
Title: "pyxform_autotesttitle"


>> XForm is invalid. See above for the errors.
org.javarosa.xform.parse.XFormParseException: Invalid calculate for the bind attached to "/pyxform_autotestname/bad" : Couldn't understand the expression starting at this point: ?$(myField)='1' in expression $(myField)='1'
	at org.javarosa.xform.parse.StandardBindAttributesProcessor.createBinding(StandardBindAttributesProcessor.java:110)
	at org.javarosa.xform.parse.XFormParser.processStandardBindAttributes(XFormParser.java:1686)
	at org.javarosa.xform.parse.XFormParser.parseBind(XFormParser.java:1704)
	at org.javarosa.xform.parse.XFormParser.parseModel(XFormParser.java:575)
	at org.javarosa.xform.parse.XFormParser.access$200(XFormParser.java:103)
	at org.javarosa.xform.parse.XFormParser$2$1.handle(XFormParser.java:241)
	at org.javarosa.xform.parse.XFormParser.parseElement(XFormParser.java:494)
	at org.javarosa.xform.parse.XFormParser.parseElement(XFormParser.java:503)
	at org.javarosa.xform.parse.XFormParser.parseElement(XFormParser.java:503)
	at org.javarosa.xform.parse.XFormParser.parseDoc(XFormParser.java:412)
	at org.javarosa.xform.parse.XFormParser.parse(XFormParser.java:326)
	at org.javarosa.xform.util.XFormUtils.getFormFromInputStream(XFormUtils.java:79)
	at org.opendatakit.validate.FormValidator.validate(FormValidator.java:410)
	at org.opendatakit.validate.FormValidator.validate(FormValidator.java:343)
	at org.opendatakit.validate.FormValidator.actionPerformed(FormValidator.java:254)
	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
	at javax.swing.DefaultButtonModel.setPressed(Unknown Source)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
	at java.awt.Component.processMouseEvent(Unknown Source)
	at javax.swing.JComponent.processMouseEvent(Unknown Source)
	at java.awt.Component.processEvent(Unknown Source)
	at java.awt.Container.processEvent(Unknown Source)
	at java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)

```

You may need to reduce the test string to only portions that do not include the unicode character.

Not sure if this means there is a unicode issue in ODK Validate or there is need to install unicode encodings in windows to support such tests.

**Comment by lognaturel** [comment]
Thank you, @ukanga! I'm ok with reducing the test string for now and looking into the Validate issue separately. At least we know we won't be getting an error so this is an improvement.

**Pull request merged**